src/CartesianProduct.jl

```diff
@@ -272,3 +308,70 @@ function ∈(x::AbstractVector{N}, cpa::CartesianProductArray{N, <:LazySet{N}}
     end
     return true
 end
+
+"""
+    vertices_list(cpa::CartesianProductArray{N})::Vector{Vector{N}} where N<:Real
+
+Return the list of vertices of a (polytopic) Cartesian product.
+
+### Input
+
+- `cpa` -- Cartesian product
+
+### Output
+
+A list of vertices.
+
+### Algorithm
+
+We assume that the underlying sets are polytopic.
+Then the high-dimensional set of vertices is just the Cartesian product of the
+low-dimensional sets of vertices.
+"""
+function vertices_list(cpa::CartesianProductArray{N})::Vector{Vector{N}} where N<:Real
+    # collect low-dimensional vertices lists
+    vlist_low = [vertices_list(X) for X in array(cpa)]
+
+    # create high-dimensional vertices list
+    indices_max = [length(vl) for vl in vlist_low]
+    m = prod(indices_max)
+    vlist = Vector{Vector{N}}(undef, m)
+    indices = ones(Int, length(vlist_low))
+    v = zeros(N, dim(cpa))
+    dim_start_j = 1
+    for vl in vlist_low
+        v_low = vl[1]
+        v[dim_start_j:dim_start_j+length(v_low)-1] = v_low
+        dim_start_j += length(v_low)
+    end
```

Comment:
did you forget an assignment to `vlist` here? since below the array `v` is modified before writing in `vlist`.

Reply:
No, this loop only writes the first full-dimensional vector `v`. The code below starts by overwriting the first block with the same values, true. I had written to `vlist` in an earlier version and started at `i = 2` below, but then I removed this part to keep the code shorter.

Reply:
Ok. Alternative: `v = vcat([vi[1] for vi in vlist_low]...)`.